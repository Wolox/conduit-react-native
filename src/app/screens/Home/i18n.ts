import i18next from 'i18next';

i18next.addResources('en', 'HOME', {
  NEW_ARTICLE: 'Create new article',
  APP_NAME: 'Conduit',
  FIRST_TAB: 'Your feed',
  SECOND_TAB: 'Global feed',
  EMPTY_ARTICLES: 'No articles are here... yet',
  TAGS: 'Tags'
});
