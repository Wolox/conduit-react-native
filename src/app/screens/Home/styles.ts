import { StyleSheet } from 'react-native';
import { BOTTOM_SEPARATOR_LINE } from '@constants/commonStyles';
import { SPACING } from '@constants/dimentions';

export default StyleSheet.create({
  container: {
    flex: 1
  },
  separator: {
    ...BOTTOM_SEPARATOR_LINE,
    width: '100%',
    alignItems: 'center'
  },
  indicator: {
    marginTop: SPACING.SMALL
  },
  titleEmptyArticles: {
    marginTop: SPACING.MEDIUM
  },
  title: {
    marginVertical: SPACING.SMALL
  },
  containerTagButton: {
    paddingHorizontal: SPACING.MEDIUM
  },
  tagButton: {
    width: 80,
    alignSelf: 'flex-end'
  },
  containerSelectedTags: {
    paddingHorizontal: SPACING.MEDIUM,
    flexDirection: 'row'
  }
});
