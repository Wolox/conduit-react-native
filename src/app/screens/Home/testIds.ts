export default {
  homeView: 'home-view',
  tagsButton: 'tags-button'
};
