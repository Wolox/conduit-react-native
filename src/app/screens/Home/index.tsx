import React, { useCallback, useRef, useEffect, useState } from 'react';
import { SafeAreaView, FlatList, ListRenderItem, View, ActivityIndicator, Text } from 'react-native';
import { useSelector, useDispatch } from 'react-redux';
import i18next from 'i18next';
import { State } from '@interfaces/reduxInterfaces';
import { Article } from '@interfaces/articlesInterface';
import { ListKeyExtractor } from '@interfaces/miscelanious';
import { Navigation } from '@interfaces/navigation';
import CustomText from '@components/CustomText';
import TabList from '@components/TabList';
import ConfirmationModal from '@components/ConfirmationModal';
import ScreenWithLoader from '@components/ScreenWithLoader';
import ArticleItem from '@components/ArticleItem';
import CustomButton from '@components/CustomButton';
import { THRESHOLD } from '@constants/pagination';
import ArticlesActions from '@redux/articles/actions';
import Routes from '@constants/routes';
import icSort from '@assets/icons/icSort.png';
import detailArticleStyles from '@screens/DetailArticle/styles';

import styles from './styles';
import './i18n';

function Home({ navigation }: Navigation) {
  const dispatch = useDispatch();
  const paginated = useRef(false);
  const [currentTab, setCurrentTab] = useState(0);
  const [deleteArticleModal, setDeleteArticleModal] = useState(false);
  const [, setArticleToDelete] = useState<Article | null>(null);

  const articles = useSelector<State, Article[]>(state => state.articles.articlesList?.page || []);
  const myArticles = useSelector<State, Article[]>(state => state.articles.myArticlesList.articles || []);
  const loadingMyArticles = useSelector<State, boolean>(state => state.articles.myArticlesListLoading);
  const loading = useSelector<State, boolean>(state => state.articles.articlesListLoading);
  const currentUser = useSelector((state: State) => state.auth.currentUser);
  const selectedTags = useSelector<State, string[]>(state => state.articles.selectedTags || []);

  const renderSeparator = useCallback(() => <View style={styles.separator} />, []);

  const handlePressArticle = useCallback(
    (article: Article) => navigation.navigate(Routes.DetailArticle, { article }),
    [navigation]
  );

  const handlePressTagsButton = useCallback(() => navigation.navigate(Routes.Tags), [navigation]);

  const onPressDeleteArticle = (item: Article) => {
    setDeleteArticleModal(true);
    setArticleToDelete(item);
  };

  const renderItem: ListRenderItem<Article> = useCallback(
    ({
      item,
      item: {
        author: { username }
      }
    }) => (
      <ArticleItem
        item={item}
        onPress={handlePressArticle}
        showDeleteIcon={username === currentUser?.user?.username}
        onDeletePress={onPressDeleteArticle}
      />
    ),
    [currentUser, handlePressArticle]
  );

  const keyExtractor: ListKeyExtractor<Article> = useCallback(
    ({ slug, author: { username } }, index) => `${index}-${slug}-${username}`,
    []
  );

  const renderFooter = useCallback(
    () => (
      <View style={articles.length ? styles.separator : null}>
        {loading && paginated.current && <ActivityIndicator size="large" style={styles.indicator} />}
      </View>
    ),
    [loading, paginated, articles]
  );

  const getArticles = useCallback(() => {
    dispatch(ArticlesActions.getArticles());
  }, [dispatch]);

  const getMyArticles = useCallback(async () => {
    await dispatch(ArticlesActions.getMyArticles());
  }, [dispatch]);

  const getTags = useCallback(() => {
    dispatch(ArticlesActions.getTags());
  }, [dispatch]);

  useEffect(() => {
    if (!paginated.current && !!articles.length) paginated.current = true;
  }, [loading, paginated, articles]);

  useEffect(() => {
    getTags();
    if (currentTab === 1 || !currentUser) {
      getArticles();
    } else {
      setTimeout(() => {
        getMyArticles();
      }, 300);
    }
  }, [currentTab, currentUser, getArticles, getMyArticles, getTags]);

  const handlePressTab = (index: number) => setCurrentTab(index);

  const showDeleteModal = () => {
    setDeleteArticleModal(false);
    setArticleToDelete(null);
  };
  const onConfirmDeleteAticle = () => {
    // uncomment but the EP return a 500 error
    const { slug = '' } = articleToDelete || {};
    dispatch(ArticlesActions.deleteArticle(slug));
    setDeleteArticleModal(false);
  };

  return (
    <SafeAreaView style={styles.container}>
      <ConfirmationModal
        title={i18next.t('HOME:CONFIRM_DELETE_ARTICLE')}
        showModal={deleteArticleModal}
        onCancel={showDeleteModal}
        onConfirm={onConfirmDeleteAticle}
      />
      <CustomText center big darkBlue bold style={styles.title}>
        {i18next.t('HOME:APP_NAME')}
      </CustomText>
      {currentUser && (
        <TabList
          tabs={[i18next.t('HOME:FIRST_TAB'), i18next.t('HOME:SECOND_TAB')]}
          onPressTab={handlePressTab}
        />
      )}
      <View style={styles.containerTagButton}>
        <CustomButton
          link
          icon={icSort}
          title={i18next.t('HOME:TAGS')}
          style={styles.tagButton}
          onPress={handlePressTagsButton}
        />
      </View>
      <View style={styles.containerSelectedTags}>
        {selectedTags.map((tag: string, index: number) => (
          <Text key={index} style={detailArticleStyles.tag}>
            {tag}
          </Text>
        ))}
      </View>
      {currentTab === 0 && currentUser ? (
        <>
          <ScreenWithLoader loading={loadingMyArticles && !myArticles.length} withInitialLoading={false}>
            {myArticles.length ? (
              <FlatList<Article>
                style={styles.container}
                data={myArticles}
                keyExtractor={keyExtractor}
                renderItem={renderItem}
                ItemSeparatorComponent={renderSeparator}
                showsVerticalScrollIndicator={false}
                ListFooterComponent={renderFooter}
              />
            ) : (
              <CustomText center style={styles.titleEmptyArticles}>
                {i18next.t('HOME:EMPTY_ARTICLES')}
              </CustomText>
            )}
          </ScreenWithLoader>
        </>
      ) : (
        <ScreenWithLoader
          loading={loading && !articles.length}
          refreshed={paginated.current}
          withInitialLoading={false}>
          <FlatList<Article>
            style={styles.container}
            data={articles}
            keyExtractor={keyExtractor}
            renderItem={renderItem}
            ItemSeparatorComponent={renderSeparator}
            onEndReachedThreshold={THRESHOLD}
            showsVerticalScrollIndicator={false}
            ListFooterComponent={renderFooter}
          />
        </ScreenWithLoader>
      )}
    </SafeAreaView>
  );
}

export default Home;
