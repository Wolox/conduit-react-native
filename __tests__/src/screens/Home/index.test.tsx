import 'jest-react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Home from '@screens/Home';
import MockedNavigator from '@mocks/MockedNavigator';

jest.mock('react-redux', () => {
  return {
    useSelector: jest.fn().mockImplementation(() => []),
    useDispatch: () => jest.fn()
  };
});

describe('test HOME SCREEN', () => {
  beforeEach(() => {
    jest.useFakeTimers();
  });
  test('to match with Snapshot', () => {
    const tree = renderer.create(<MockedNavigator component={Home} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
