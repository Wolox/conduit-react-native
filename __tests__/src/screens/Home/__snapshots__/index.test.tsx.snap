// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test HOME SCREEN to match with Snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                aria-level="1"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                MockedScreen
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <RNSScreenContainer
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        activityState={2}
        forwardedRef={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <RCTSafeAreaView
                      emulateUnlessSupported={true}
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <Modal
                        animationType="fade"
                        hardwareAccelerated={false}
                        transparent={true}
                        visible={false}
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#00000033",
                              "flex": 1,
                              "justifyContent": "center",
                              "marginTop": 22,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#FFF",
                                "borderRadius": 20,
                                "elevation": 5,
                                "height": 180,
                                "margin": 20,
                                "padding": 35,
                                "shadowColor": "#555555",
                                "shadowOffset": Object {
                                  "height": 2,
                                  "width": 0,
                                },
                                "shadowOpacity": 0.3,
                                "shadowRadius": 4,
                                "width": "93%",
                              }
                            }
                          >
                            <Text
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "transparent",
                                    "color": "#000",
                                    "fontFamily": "Nunito",
                                    "fontSize": 25.142857142857142,
                                    "fontStyle": "normal",
                                    "fontWeight": "400",
                                  },
                                  Array [
                                    Object {
                                      "color": "#000",
                                      "fontFamily": "Nunito",
                                      "fontSize": 25.142857142857142,
                                      "fontStyle": "normal",
                                      "fontWeight": "700",
                                    },
                                  ],
                                  Object {
                                    "marginBottom": 15,
                                    "textAlign": "center",
                                  },
                                ]
                              }
                            >
                              HOME:CONFIRM_DELETE_ARTICLE
                            </Text>
                            <View
                              style={
                                Object {
                                  "flexDirection": "row",
                                  "justifyContent": "space-evenly",
                                  "marginTop": 20,
                                  "width": "100%",
                                }
                              }
                            >
                              <View
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onBlur={[Function]}
                                onClick={[Function]}
                                onFocus={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Array [
                                    Object {
                                      "borderRadius": 20,
                                      "elevation": 2,
                                      "padding": 15,
                                    },
                                    Object {
                                      "backgroundColor": "#5CB85C",
                                    },
                                  ]
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "color": "#FFF",
                                      "fontWeight": "bold",
                                      "textAlign": "center",
                                    }
                                  }
                                >
                                  CONFIRMATION_MODAL:CANCEL
                                </Text>
                              </View>
                              <View
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onBlur={[Function]}
                                onClick={[Function]}
                                onFocus={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Array [
                                    Object {
                                      "borderRadius": 20,
                                      "elevation": 2,
                                      "padding": 15,
                                    },
                                    Object {
                                      "backgroundColor": "#FF2400",
                                    },
                                  ]
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "color": "#FFF",
                                      "fontWeight": "bold",
                                      "textAlign": "center",
                                    }
                                  }
                                >
                                  CONFIRMATION_MODAL:DELETE
                                </Text>
                              </View>
                            </View>
                          </View>
                        </View>
                      </Modal>
                      <Text
                        style={
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "color": "#000",
                              "fontFamily": "Nunito",
                              "fontSize": 25.142857142857142,
                              "fontStyle": "normal",
                              "fontWeight": "400",
                            },
                            Array [
                              Object {
                                "color": "#000",
                                "fontFamily": "Nunito",
                                "fontSize": 25.142857142857142,
                                "fontStyle": "normal",
                                "fontWeight": "700",
                              },
                              Object {
                                "textAlign": "center",
                              },
                              Object {
                                "color": "#36424a",
                              },
                              Object {
                                "fontSize": 31.428571428571427,
                              },
                            ],
                            Object {
                              "marginVertical": 12,
                            },
                          ]
                        }
                      >
                        HOME:APP_NAME
                      </Text>
                      <View
                        style={
                          Object {
                            "flexDirection": "row",
                            "paddingHorizontal": 16,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "width": "50%",
                            }
                          }
                        >
                          <View
                            accessible={true}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "borderBottomColor": "#5CB85C",
                                "borderBottomWidth": 3,
                                "opacity": 1,
                              }
                            }
                          >
                            <Text
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "transparent",
                                    "color": "#000",
                                    "fontFamily": "Nunito",
                                    "fontSize": 25.142857142857142,
                                    "fontStyle": "normal",
                                    "fontWeight": "400",
                                  },
                                  Array [
                                    Object {
                                      "textAlign": "center",
                                    },
                                    Object {
                                      "color": "#5CB85C",
                                    },
                                  ],
                                  Object {
                                    "margin": 10,
                                  },
                                ]
                              }
                            >
                              HOME:FIRST_TAB
                            </Text>
                          </View>
                        </View>
                        <View
                          style={
                            Object {
                              "width": "50%",
                            }
                          }
                        >
                          <View
                            accessible={true}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "borderBottomColor": "#E5E5E5",
                                "borderBottomWidth": 3,
                                "opacity": 1,
                              }
                            }
                          >
                            <Text
                              style={
                                Array [
                                  Object {
                                    "backgroundColor": "transparent",
                                    "color": "#000",
                                    "fontFamily": "Nunito",
                                    "fontSize": 25.142857142857142,
                                    "fontStyle": "normal",
                                    "fontWeight": "400",
                                  },
                                  Array [
                                    Object {
                                      "textAlign": "center",
                                    },
                                  ],
                                  Object {
                                    "margin": 10,
                                  },
                                ]
                              }
                            >
                              HOME:SECOND_TAB
                            </Text>
                          </View>
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "paddingHorizontal": 16,
                          }
                        }
                      >
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "alignSelf": "flex-end",
                              "backgroundColor": "transparent",
                              "borderColor": "#5CB85C",
                              "borderWidth": 0,
                              "flexDirection": "row",
                              "justifyContent": "center",
                              "opacity": 1,
                              "padding": 8,
                              "width": 80,
                            }
                          }
                        >
                          <Image
                            resizeMode="contain"
                            source={
                              Object {
                                "process": [Function],
                              }
                            }
                            style={
                              Array [
                                Object {
                                  "height": 20,
                                  "width": 20,
                                },
                                undefined,
                                Object {
                                  "tintColor": "#5CB85C",
                                },
                              ]
                            }
                          />
                          <Text
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "color": "#000",
                                  "fontFamily": "Nunito",
                                  "fontSize": 25.142857142857142,
                                  "fontStyle": "normal",
                                  "fontWeight": "400",
                                },
                                Array [],
                                Array [
                                  Array [],
                                  undefined,
                                  Object {
                                    "color": "#5CB85C",
                                  },
                                ],
                              ]
                            }
                          >
                            HOME:TAGS
                          </Text>
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "flexDirection": "row",
                            "paddingHorizontal": 16,
                          }
                        }
                      />
                      <View
                        style={
                          Object {
                            "backgroundColor": "#FFF",
                            "flex": 1,
                            "justifyContent": "center",
                          }
                        }
                      >
                        <ActivityIndicator
                          animating={true}
                          color="#288CCB"
                          hidesWhenStopped={true}
                          size="large"
                        />
                      </View>
                    </RCTSafeAreaView>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenContainer>
  </RNCSafeAreaProvider>
</View>
`;
